### PR TITLE
[ContentCacheManager] add support for stream detection across chunks

### PR DESCRIPTION
**Pull request by das7pad**
### Description

For https://github.com/overleaf/issues/issues/3871
For https://github.com/overleaf/clsi/pull/224#issuecomment-839883764

This PR is adding support for stream detection across chunks. 

#### Related Issues / PRs

For https://github.com/overleaf/issues/issues/3871
For https://github.com/overleaf/clsi/pull/224#issuecomment-839883764

### Review

Retain a small part (6 or 9 bytes) of each chunk in memory for providing
 the next iteration with enough context for finding the start/end marker
 of a range.

I've also added constants for the markers.

#### Potential Impact

Low.

#### Manual Testing Performed

```
stream = [
    Buffer.from("abcstr"),
    Buffer.from("eam123endstreamABC"),
    Buffer.from("str"),
    Buffer.from("eam(|"),
    Buffer.from(")end"),
    Buffer.from("stream-_~stream!$%endstream"),
  ]
```

### CODE REVIEW

**Review by briangough — COMMENTED** [review]
I think this is a case where a unit test would be useful.

File: app/js/ContentCacheManager.js [hunk 1]
```diff
@@ -41,15 +46,17 @@ class PdfStreamsExtractor {
     this.inStream = false
     this.streamStartIndex = 0
     this.buffers = []
+    this.lastChunk = Buffer.alloc(0)
   }
 
   consume(chunk) {
     let chunkIndex = 0
     const pdfStreams = []
+    chunk = Buffer.concat([this.lastChunk, chunk])
```
Comment (briangough) [inline app/js/ContentCacheManager.js:55]: If we need more efficiency we can do this without needing to concat the buffers, by handling the overlap as a separate case.  Let's leave that for later though.

**Comment by das7pad** [comment]
> I think this is a case where a unit test would be useful.

Indeed! Now added via a force-push. It also surfaced a bug with the `fileIndex` advancing. Thanks for asking for it.

**Review by briangough — APPROVED** [review]
That's great.

**Pull request merged**